prov/efa/src/efa_hmem.c

```diff
@@ -276,70 +227,69 @@ static int efa_domain_hmem_info_init_neuron(struct efa_domain *efa_domain)
 		EFA_WARN(FI_LOG_DOMAIN,
 			 "Failed to deregister Neuron buffer: %s\n",
 			 fi_strerror(-ret));
-		return ret;
+		return;
 	}
 
 	info->p2p_supported_by_device = true;
-	efa_domain_hmem_info_init_protocol_thresholds(efa_domain, FI_HMEM_NEURON);
-	if (-FI_ENODATA != fi_param_get(&efa_prov, "inter_max_medium_message_size", &tmp_value)) {
-		EFA_WARN(FI_LOG_DOMAIN,
-		         "The environment variable FI_EFA_INTER_MAX_MEDIUM_MESSAGE_SIZE was set, "
-		         "but EFA HMEM via Neuron API only supports eager and runting read protocols. "
-				 "The variable will not modify Neuron memory run config.\n");
-	}
-
+	return;
 #endif
-	return 0;
+	return;
 }
 
 /**
- * @brief          Initialize the efa_hmem_info state for FI_HMEM_SYNAPSEAI
+ * @brief Initialize the efa_hmem_info state for iface
  *
  * @param[in,out]  efa_domain  Pointer to struct efa_domain
- *
- * @return         0
+ * @param[in]      iface       HMEM interface
  */
-static int efa_domain_hmem_info_init_synapseai(struct efa_domain *efa_domain)
+static void
+efa_domain_hmem_info_init_iface(struct efa_domain *efa_domain, enum fi_hmem_iface iface)
 {
-#if HAVE_SYNAPSEAI
-	size_t tmp_value;
-	struct efa_hmem_info *info = &efa_domain->hmem_info[FI_HMEM_SYNAPSEAI];
+	struct efa_hmem_info *info = &efa_domain->hmem_info[iface];
 
-	if (!ofi_hmem_is_initialized(FI_HMEM_SYNAPSEAI)) {
-		EFA_INFO(FI_LOG_DOMAIN, "FI_HMEM_SYNAPSEAI is not initialized\n");
-		return 0;
+	if (!ofi_hmem_is_initialized(iface)) {
+		EFA_INFO(FI_LOG_DOMAIN, "%s is not initialized\n",
+		         fi_tostr(&iface, FI_TYPE_HMEM_IFACE));
+		return;
 	}
 
-	if (!(g_device_list[0].device_caps & EFADV_DEVICE_ATTR_CAPS_RDMA_READ)) {
+	if ((iface == FI_HMEM_SYNAPSEAI || iface == FI_HMEM_NEURON) &&
+	    !(g_device_list[0].device_caps & EFADV_DEVICE_ATTR_CAPS_RDMA_READ)) {
```

Comment:
Nit: you can reuse https://github.com/ofiwg/libfabric/blob/main/prov/efa/src/efa_device.c#L243 `efa_device_support_rdma_read`